journalsmanager/utils.py

```diff
@@ -145,25 +146,62 @@ def export_issns(rows):
 def export_to_autocomplete(rows):
     data = []
     try:
+        canonicalBibs = []
+        bibcodeFile = config.get('CANONICAL_BIBS', None)
+        if bibcodeFile:
+            bibcodeFile = JDB_DATA_DIR + '/' + bibcodeFile
+            with open(bibcodeFile, 'r') as fb:
+                for l in fb.readlines():
+                    canonicalBibs.append(l.strip())
+        cites = {}
+        citationFile = config.get('CITATION_COUNTS', None)
+        if citationFile:
+            citationFile = JDB_DATA_DIR + '/' + citationFile
+            with open(citationFile, 'r') as fc:
+                for l in fc.readlines():
+                    (bibcode, tcit, rcit) = l.strip().split('\t')
+                    if cites.get(bibcode, None):
+                        cites[bibcode] += (int(tcit) + int(rcit))
+                    else:
+                        cites[bibcode] = (int(tcit) + int(rcit))
         for r in rows:
             bibstem = r.get('bibstem', None)
+            if len(bibstem) < 5:
+                bibstem = bibstem.ljust(5, '.')
             names = list()
-            if r.get('name', None):
-                names.append(r.get('name', None))
-            if r.get('translated_name', None):
-                names.append(r.get('translated_name', None))
-            if r.get('native_name', None):
-                names.append(r.get('native_name', None))
-            if r.get('transliterated_name', None):
-                names.append(r.get('transliterated_name', None))
-            if bibstem and names:
-                data.append({'value': bibstem, 'label': names})
-            elif not bibstem:
-                print('what the hell? %s' % str(r))
-        result = {'data': data}
+            bibcodeList = [c for c in canonicalBibs if bibstem in c]
```

Comment:
If bibstem is JGR, this list comprehension finds all bibcodes with bibstems JGR, JGRA, JGRB, ... (same with any other bibstem that has partial matches in other bibstems)

Reply:
In L169-170 I do a conversion of all bibstems with len < 5 to pad them with trailing dots, so it's doing a search on 'JGR..' rather than 'JGR' in the list comprehension.

Reply:
That's what I figured in hindsight :-) So, forget about this comment!